devtools/bom-descriptor-json/src/main/resources/catalog-overrides.json

```diff
@@ -146,6 +145,11 @@
   "metadata":{
     "project": {
       "default-codestart": "rest",
+      "codestart-data": {
+        "dockerfile.jvm.from": "registry.access.redhat.com/ubi9/openjdk-${recommended-java-version}:1.21",
```

Comment:
I think we are a bit more clever than that in the code:

```
    public static String getDefaultJvmImage(CompiledJavaVersionBuildItem.JavaVersion version) {
        switch (version.isJava21OrHigher()) {
            case TRUE:
                return UBI9_JAVA_21;
            default:
                return UBI9_JAVA_17;
        }
    }
```

We might have to have a pattern that we replace dynamically. Not sure how it will fly though.